BezierKit/Library/PathComponent+Arc.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  PathComponent+Arc.swift
+//  BezierKit
+//
+//  Created by ky1vstar on 15.10.2020.
+//  Copyright © 2020 Holmes Futrell. All rights reserved.
+//
+
+import Foundation
+
+extension PathComponent {
+    public convenience init(
+        ovalIn rect: CGRect,
+        numberOfCurves: Int = 4
+    ) {
+        let diameter = min(rect.width, rect.height)
+        let radius = diameter / 2
+        
+        let tempPath = PathComponent(
+            arcCenter: .init(x: radius, y: radius),
+            radius: radius,
+            startAngle: 0,
+            endAngle: 2 * .pi,
+            clockwise: true
+            numberOfCurves: numberOfCurves
+        )
+        
+        let transform = CGAffineTransform.identity
+            .translatedBy(x: rect.origin.x, y: rect.origin.y)
+            .scaledBy(
+                x: rect.width / diameter,
```

Comment:
I think that if the rect is zero sized this will cause a divide by zero. It's an edge case, but actually quite likely to occur in practice. We should check how `CGPath(ellipseIn:)` handles zero sized rects and use the same fallback behavior.